### PR TITLE
Expose endpoint for Load Testing

### PR DESCRIPTION
**Pull request by kajan-bolt**
# Description
This PR describes how to expose an endpoint that allows us to trigger the part of the code that is run when a user goes to the cart page (or is on any page that has a mini-cart). This endpoint adds given items to a cart, creates a bolt order, and then return the order token and order reference.

This is going to be used to Load Test the plugin. Since this endpoint is only for LoadTesting, I put all the relevant files in the Test Directory and made a README file that describes how to setup the endpoint. 

Fixes: https://app.asana.com/0/941920570700290/1149771740816849/f

(Optional: Add a changelog by writing a comment after "#changelog" on the next line)

\#changelog

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [ x ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [ x ] Successfully tested locally (or docker image)
- [ x ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# Checklist:

- [ ] My code follows the style guidelines of this project.
- [ x ] I have performed a self-review of my own code.
- [ x ] I have commented my code, particularly in hard-to-understand areas.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have created or modified unit tests to sufficiently cover my changes.
- [ ] I have added my Asana task link and provided a changelog message if applicable.

### CODE REVIEW

File: Test/Load/setup-files/LoadTestCartData.php [hunk 1]
```diff
@@ -0,0 +1,217 @@
+<?php
```
Comment (roopakv) [inline Test/Load/setup-files/LoadTestCartData.php]: At a high level this seems to only load test this endpoint? What are we hoping that load testing this test endpoint which is not used anywhere will give us? I might be missing some context.

Wouldn't a better load test be to extract some of the endpoints we actually have and load test them?

Comment (kajan-bolt) [reply Test/Load/setup-files/LoadTestCartData.php]: You are right, and we will be testing the other endpoints of the plugin, but I split the actual tests into a separate PR found here: https://github.com/BoltApp/bolt-magento2/pull/429

This PR is just to expose this part of the plugin that is triggered each time we get to the cart page (i.e `getBoltpayOrder()` which converts the cart on a merchant site to a bolt order ). 

But, for more context the load tests found in the other PR will test the user flow through checkout: 
- Getting to the cart page (which creates the order token and order reference)
- Calling the shipping and tax web hook set up on the merchant's site
- Calling the Pre-auth web hook set up on the merchant's site 

The purpose behind these load tests is to see whether our plugin is the actual problem when a merchant has a significant number of users accessing their store.

File: Test/Load/setup-files/README.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+2. Configure your M2 store with Bolt: [M2 Bolt Configuration](https://docs.bolt.com/docs/magento-2-integration#section-2-plugin-configuration)
+
+3. Add the line below in `bolt-magento2/etc/di.xml`: 
+	```
+	<preference for="Bolt\Boltpay\Api\LoadTestCartDataInterface" type="Bolt\Boltpay\Model\Api\LoadTestCartData" />
+	```
+	Add it below the comment: 
+	```
+	<!-- For rest api hook integration -->
```
Comment (matt-thomason) [inline Test/Load/setup-files/README.md]: Was this commented added in a PR? I must've missed it

Comment (kajan-bolt) [reply Test/Load/setup-files/README.md]: I'm not to sure which PR it was added in but I can see it in master: https://github.com/BoltApp/bolt-magento2/blob/master/etc/di.xml#L37

File: Test/Load/setup-files/LoadTestCartData.php [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+                throw new BoltException(
+                    __('Missing cart data.'),
+                    null,
+                    self::E_BOLT_GENERAL_ERROR
+                );
+            }
+
+            $this->preProcessWebhook();
+
+            // Create cart with the given products
```
Comment (njfaries) [inline Test/Load/setup-files/LoadTestCartData.php]: For my own understanding here, you say you're creating a cart with given products, but it looks like you're creating a quote from the cart variable passed in. Is this just naming convention in m2?

Comment (kajan-bolt) [reply Test/Load/setup-files/LoadTestCartData.php]: Yeah so I believe in M2 a quote is similar to a cart. And so we just add the items to the quote.

**Pull request merged**